lib/scholar/covariance/utils.ex

```diff
@@ -0,0 +1,39 @@
+defmodule Scholar.Covariance.Utils do
+  @moduledoc false
+  import Nx.Defn
+  require Nx
+
+  defn center(x, assume_centered \\ false) do
+    x =
+      case Nx.shape(x) do
+        {_} -> Nx.new_axis(x, 1)
+        _ -> x
+      end
+
+    location =
+      if assume_centered do
+        0
+      else
+        Nx.mean(x, axes: [0])
+      end
+
+    {x - location, location}
+  end
+
+  defn empirical_covariance(x) do
```

Comment:
You should be able to use `Nx.covariance/2` instead.

Reply:
I don't think so, look at the PR description

Reply:
Right, I forgot to ask: @norm4nn did you try setting `ddof: 0` when calling `Nx.covariance/2`?

Reply:
Sorry, `ddof: 0` is default. It's strange that the tests are failing, because `Nx.covariance/2` does exactly what is implemented here. Could it be the case that the data in your test is not centered and you are using `Nx.covariance/2`?

Reply:
Yeah, you are right! This is the case, I will fix this on Thursday.